src/unix/linux_like/android/mod.rs

```diff
@@ -1020,6 +1034,31 @@ cfg_if! {
                     .finish()
             }
         }
+
+        impl Eq for ethhdr {}
+
+        impl PartialEq for ethhdr {
```

Comment:
This `PartialEq` definition appears to lack a check on the `h_proto` field.

Reply:
I did not forget but until I solve the alignment issue in some platforms and as I was playing with the h_proto type I do not bother with it just yet (still a draft).